### PR TITLE
ref(streams): Add support for managed consumer methods

### PR DESCRIPTION
**Pull request by tkaemming**
This adds the `assign` and `unassign` methods to `KafkaConsumer`. See docstrings.

### CODE REVIEW

File: snuba/utils/streams/consumer.py [hunk 1]
```diff
@@ -173,21 +172,44 @@ def __init__(
...
+            if item.offset >= 0:
+                result[partition] = item.offset
+            elif item.offset == OFFSET_INVALID:
+                result[partition] = self.__resolve_partition_starting_offset(item)
+            else:
+                raise ValueError("received unexpected offset")
+
+        return result
+
+    # Balanced Consumer Methods
```
Comment (fpacifici) [inline snuba/utils/streams/consumer.py]: Not sure what you mean by balanced consumer here.

File: snuba/utils/streams/consumer.py [hunk 2]
```diff
@@ -290,6 +303,63 @@ def unsubscribe(self) -> None:
...
+                ]
+            ),
+        }
+
+        self.__consumer.assign(
+            [
+                ConfluentTopicPartition(partition.topic.name, partition.index, offset)
+                for partition, offset in assignment.items()
+            ]
+        )
```
Comment (fpacifici) [inline snuba/utils/streams/consumer.py]: What happens if I try to assign an offset lower than `earliest` or higher than `latest` ? Is it suppsoed to throw or to let Confluent handle this case ?

File: snuba/utils/streams/consumer.py (hunk 1 above)
Comment (fpacifici) [reply snuba/utils/streams/consumer.py]: ok, if it is a term from pykafka I think we are fine

**Review by fpacifici — APPROVED** [review]

**Comment by tkaemming** [comment]
This ended up not being needed right now. We'll probably need to reconsider implementing this in some fashion later when there are a large number of streaming subscription updates or commit log entries, but we can cross that bridge when we come to it.

**Pull request closed without merging**